config/src/main/java/com/alibaba/fescar/config/FileConfiguration.java

```diff
@@ -123,6 +124,7 @@ public boolean removeConfig(String dataId, long timeoutMills) {
     public void addConfigListener(String dataId, ConfigChangeListener listener) {
         configListenersMap.putIfAbsent(dataId, new ArrayList<ConfigChangeListener>());
         configListenersMap.get(dataId).add(listener);
+        listenedConfigMap.putIfAbsent(dataId, getConfig(dataId));
```

Comment:
maybe put?

Reply:
if have many listeners and  this value has been updated. the data is old value

Reply:
if exists listener, the value in listenedconfigmap will be update in the ConfigChangeRunnable,

Reply:
> if exists listener, the value in listenedconfigmap will be update in the ConfigChangeRunnable,

yes u are right